service/history/api/multioperation/api.go

```diff
@@ -131,15 +131,26 @@ func Invoke(
 
 	// For workflow id conflict policy terminate-existing, always attempt a start
 	// since that works when the workflow is already running *and* when it's not running.
-	if startReq.StartRequest.WorkflowIdConflictPolicy == enumspb.WORKFLOW_ID_CONFLICT_POLICY_TERMINATE_EXISTING {
-		if resp, outcome, err := startAndUpdateWorkflow(
-			ctx, shardContext, workflowConsistencyChecker, starter, updater,
-		); err != nil {
+	conflictPolicy := startReq.StartRequest.WorkflowIdConflictPolicy
+	if conflictPolicy == enumspb.WORKFLOW_ID_CONFLICT_POLICY_TERMINATE_EXISTING {
+		resp, startOutcome, err := startAndUpdateWorkflow(ctx, starter, updater)
+		if err != nil {
 			return nil, err
-		} else if outcome != startworkflow.StartDeduped {
+		}
+
+		switch startOutcome {
+		case startworkflow.NoStart:
```

Comment:
Why do we need the state startworkflow.NoStart? I see that it is returned only when there is error. If that that is the case, isn't it redundant to return this?

Similarly for startReused like here: https://github.com/temporalio/temporal/blob/7c9e7c1aa4cfe2afd0a542c2b4ab7053cbc9958d/service/history/api/startworkflow/api.go#L423-L423

Is it possible to include this info in the error type itself?

Reply:
It is 100% redundant. But I have to return _something_ for the "enum". Originally I had just `0`, but Alex encouraged me to give it an explicit name. And the reason I put it in the switch is that it's the only way to make the linter not complain (the alternative of using `default` is dangerous IMO).